### ISSUE TITLE
[BUG] invalid memory address or nil pointer dereference / serverless-init / Open Telemetrie

### ISSUE THREAD

**Issue by mhubig** [DataDog/datadog-agent#31126]
**Agent Environment**

- Datadog Agent from here `COPY --from=datadog/serverless-init:1.5.1 /datadog-init /app/datadog-init`
- Runs within a GCP CloudRun Service like this: `ENTRYPOINT ["/app/datadog-init"]`

**Configuration**

```Dockerfile
ENV DD_SITE=datadoghq.eu
ENV DD_TRACE_ENABLED=true
ENV DD_LOGS_ENABLED=true
ENV DD_LOG_LEVEL=error
ENV DD_SOURCE=java
ENV DD_TRACE_PROPAGATION_STYLE=datadog
ENV DD_LOGS_INJECTION=true
ENV DD_JMXFETCH_ENABLED=true
ENV DD_HTTP_CLIENT_TAG_QUERY_STRING=true
ENV DD_HTTP_SERVER_TAG_QUERY_STRING=true
ENV DD_OTLP_CONFIG_RECEIVER_PROTOCOLS_GRPC_ENDPOINT=127.0.0.1:4317
ENV DD_OTLP_CONFIG_RECEIVER_PROTOCOLS_HTTP_ENDPOINT=127.0.0.1:4318

ENV DD_VERSION=${VERSION}
ENV DD_SERVICE_MAPPING=jms:tibco,postgresql:foo-db
ENV DD_TAGS=scope:runners,team:runners
ENV DD_GIT_COMMIT_SHA=${DD_GIT_COMMIT_SHA}
ENV DD_GIT_REPOSITORY_URL=${DD_GIT_REPOSITORY_URL}
ENV DD_SERVICE=foo-import-store-availabilities
```

**Describe what happened:**

Thing crashes like this:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xa8 pc=0x1f66e8a]

goroutine 61 [running]:
github.com/DataDog/datadog-agent/comp/otelcol/otlp.(*tagEnricher).Enrich(0xc000af91b8, {0x0?, 0x0?}, {0x0, 0x0, 0x41d965?}, 0xc00086f500)
	/tmp/dd/datadog-agent/comp/otelcol/otlp/collector.go:77 +0x18a
github.com/DataDog/datadog-agent/comp/otelcol/otlp/components/exporter/serializerexporter.(*serializerConsumer).ConsumeTimeSeries(0xc0006c8a00, {0x29da788?, 0xc0003ca000?}, 0xc00086f500, 0x0, 0x180825a62a64be80, 0x41687919dc000000)
	/tmp/dd/datadog-agent/comp/otelcol/otlp/components/exporter/serializerexporter/consumer.go:78 +0x91
github.com/DataDog/opentelemetry-mapping-go/pkg/otlp/metrics.(*Translator).mapNumberMetrics(0xc000538ea0, {0x29da788, 0xc0003ca000}, {0x7fd5d6652cd8, 0xc0006c8a00}, 0xc000c79878, 0x0, {0xc000283de8?, 0xc000f91650?})
	/root/go/pkg/mod/github.com/!data!dog/opentelemetry-mapping-go/pkg/otlp/metrics@v0.21.0/metrics_translator.go:172 +0x208
github.com/DataDog/opentelemetry-mapping-go/pkg/otlp/metrics.(*Translator).mapToDDFormat(0xc000538ea0, {0x29da788, 0xc0003ca000}, {0xc00104e008?, 0xc000f91650?}, {0x29c2508, 0xc0006c8a00}, {0xc0005ace80, 0x1, 0x4}, ...)
	/root/go/pkg/mod/github.com/!data!dog/opentelemetry-mapping-go/pkg/otlp/metrics@v0.21.0/metrics_translator.go:837 +0x1225
github.com/DataDog/opentelemetry-mapping-go/pkg/otlp/metrics.(*Translator).MapMetrics(0xc000538ea0, {0x29da788, 0xc0003ca000}, {0xc000283dd0?, 0xc000f91650?}, {0x29c2508, 0xc0006c8a00})
	/root/go/pkg/mod/github.com/!data!dog/opentelemetry-mapping-go/pkg/otlp/metrics@v0.21.0/metrics_translator.go:813 +0xb54
github.com/DataDog/datadog-agent/comp/otelcol/otlp/components/exporter/serializerexporter.(*Exporter).ConsumeMetrics(0xc0009943c0, {0x29da788, 0xc0003ca000}, {0xc000283dd0?, 0xc000f91650?})
	/tmp/dd/datadog-agent/comp/otelcol/otlp/components/exporter/serializerexporter/exporter.go:186 +0x105
go.opentelemetry.io/collector/exporter/exporterhelper.(*metricsRequest).Export(0x0?, {0x29da788?, 0xc0003ca000?})
	/root/go/pkg/mod/go.opentelemetry.io/collector/exporter@v0.111.0/exporterhelper/metrics.go:61 +0x31
go.opentelemetry.io/collector/exporter/exporterhelper/internal.(*TimeoutSender).Send(0xc000ed4000?, {0x29da788?, 0xc0003ca000?}, {0x29c3798?, 0xc0006fbcf8?})
	/root/go/pkg/mod/go.opentelemetry.io/collector/exporter@v0.111.0/exporterhelper/internal/timeout_sender.go:45 +0x48
go.opentelemetry.io/collector/exporter/exporterhelper/internal.(*BaseRequestSender).Send(0xc0000ea7d0?, {0x29da788?, 0xc0003ca000?}, {0x29c3798?, 0xc0006fbcf8?})
	/root/go/pkg/mod/go.opentelemetry.io/collector/exporter@v0.111.0/exporterhelper/internal/request_sender.go:28 +0x30
go.opentelemetry.io/collector/exporter/exporterhelper.(*metricsSenderWithObservability).Send(0xc000660b40, {0x29da7f8?, 0xc0006fc0a0?}, {0x29c3798, 0xc0006fbcf8})
	/root/go/pkg/mod/go.opentelemetry.io/collector/exporter@v0.111.0/exporterhelper/metrics.go:161 +0x96
go.opentelemetry.io/collector/exporter/exporterhelper/internal.(*BaseRequestSender).Send(0x0?, {0x29da7f8?, 0xc0006fc0a0?}, {0x29c3798?, 0xc0006fbcf8?})
	/root/go/pkg/mod/go.opentelemetry.io/collector/exporter@v0.111.0/exporterhelper/internal/request_sender.go:28 +0x30
go.opentelemetry.io/collector/exporter/exporterhelper/internal.NewQueueSender.func1({0x29da7f8?, 0xc0006fc0a0?}, {0x29c3798, 0xc0006fbcf8})
	/root/go/pkg/mod/go.opentelemetry.io/collector/exporter@v0.111.0/exporterhelper/internal/queue_sender.go:90 +0x7d
```

**Describe what you expected:**

Thing not crashing

**Comment by mhubig** [comment]
I managed to fix the thing by downgrading to version 1.4.0 ...

**Comment by mhubig** [comment]
This has lead to several production problem over the weekend on our side do to the agent crashes.

**Comment by duncanpharvey** [comment]
Hi @mhubig,

Thanks for bringing this to our attention! We're currently looking into a fix.

Thanks,
Duncan

**Comment by mhubig** [comment]
Hi there, any updates?

**Comment by nina9753** [comment]
HI sorry for the delay, we recently released a new version of [serverless-init-1.5.2 ](https://hub.docker.com/repository/docker/datadog/serverless-init/general)with a fix for the otpl error. Can you please try updating to the latest version.
